src/modules/Kdump.rb

```diff
@@ -1137,27 +1124,18 @@ def crash_kernel_values
       # modification.
       # The old value (ensuring the Array format) will be returned.
       if @crashkernel_list_ranges
-        if @crashkernel_param_values.is_a? Symbol
-          return Array(@crashkernel_param_values.to_s)
-        else
-          return Array(@crashkernel_param_values.dup)
-        end
+        return Array(@crashkernel_param_values.to_s) if @crashkernel_param_values.is_a?(Symbol)
+
+        return Array(@crashkernel_param_values.dup)
       end
 
       result = []
       high = @allocated_memory[:high]
-      if high && high.to_i != 0
-        result << high + "M,high"
-      end
+      result << high + "M,high" if high && high.to_i != 0
       low = @allocated_memory[:low]
-      if low && low.to_i != 0
-        # Add the ',low' suffix only there is a ',high' one
-        if result.empty?
-          result << "#{low}M"
-        else
-          result << "#{low}M,low"
-        end
-      end
+      # Add the ',low' suffix only there is a ',high' one
+      result << (result.empty? ? "#{low}M" : "#{low}M,low") if low && low.to_i != 0
```

Comment:
Quite complex. Sincerely, I would split it

```ruby
...
low_prefix = result.empty? ? "#{low}M" : "#{low}M,low"
result << low_prefix if low && !ow.to_i != 0Ç
...
```

Reply:
this way on other hand you use `low` which can be nil, which does not crash, but is a bit strange.